### PR TITLE
fix(synchronise_tracking_and_event_data): fix check for missing playe…

### LINKED ISSUE #78: Missing `player_id` in Opta Event prevents synchronisation
I came across a match in which one of the `foul` (`type_id = 4`) Opta f24 events is missing the `player_id` attribute:

```
    <Event id="2539128075" event_id="552" type_id="4" period_id="2" min="61" sec="22" team_id="320" outcome="1" x="15.6" y="87.8" timestamp="2023-04-25T15:00:13.260" timestamp_utc="2023-04-25T14:00:13.260" last_modified="2023-04-28T10:10:14" version="1682673014251">
      <Q id="4219535657" qualifier_id="233" value="612" />
      <Q id="4219535623" qualifier_id="10" />
      <Q id="4219535629" qualifier_id="285" />
      <Q id="4219535625" qualifier_id="56" value="Back" />
      <Q id="4219535627" qualifier_id="152" />
    </Event>
```

The method `match.player_id_to_column_id` (called in `_create_sim_mat`) ends up throwing an error as the `player_id` (set to `-999` in `get_match`) cannot be matched with any of the tracking columns.

The error is handled correctly but I believe the intention was not to stop synchronising but rather to use the fallback `player_ball_diff = 0` for the similarity matrix.

### PR DESCRIPTION
**Pull request by rdghe**
closes #78

### CODE REVIEW

**Comment by rdghe** [comment]
Maybe worth considering making `-999` a global variable for missing data defined somewhere, as currently it's "hardcoded" in a bunch of places. But that's out of the scope of the issue at hand.

**Review by Alek050 — CHANGES_REQUESTED** [review]
Thanks for pointing this out. Make sure to also add a test for this so we are certain that this will not be changed in future refactors.

**Comment by rdghe** [comment]
Tests added @Alek050

File: tests/utils/test_synchronise_tracking_and_event_data.py [hunk 1]
```diff
@@ -162,12 +162,45 @@ def test_create_sim_mat_without_player(self):
...
         event_data.loc[2, "player_id"] = np.nan
+        _assert_sim_mats_equal(tracking_data, event_data)
 
-        res = _create_sim_mat(
-            tracking_batch=tracking_data,
-            event_batch=event_data,
-            match=self.match_to_sync,
-        )
+        # Test with player_id = -999
+        event_data.loc[2, "player_id"] = -999
```
[the remark this answers is not in the document]
Comment (rdghe) [reply tests/utils/test_synchronise_tracking_and_event_data.py:179]: ++

File: tests/utils/test_synchronise_tracking_and_event_data.py [hunk 2]
```diff
@@ -162,12 +162,45 @@ def test_create_sim_mat_without_player(self):
         tracking_data.reset_index(inplace=True)
         event_data = self.match_to_sync.event_data
```
[the remark this answers is not in the document]
Comment (rdghe) [reply tests/utils/test_synchronise_tracking_and_event_data.py:163]: Done, but I suppose it makes sense to have this everywhere `event_data` or `tracking_data` is accessed from `self.match_to_sync`, so I did that. Let me know if I am wrong

File: databallpy/utils/synchronise_tracking_and_event_data.py [hunk 3]
```diff
@@ -186,7 +186,7 @@ def _create_sim_mat(
             tracking_batch["ball_y"] - event["start_y"],
         )
 
-        if not np.isnan(event["player_id"]):
+        if not np.isnan(event["player_id"]) and event["player_id"] != -999:
```
[the remark this answers is not in the document]
Comment (rdghe) [reply databallpy/utils/synchronise_tracking_and_event_data.py:189]: ++

**Review by Alek050 — APPROVED** [review]

**Pull request merged**